probe-rs/src/architecture/arm/memory/adi_v5_memory_interface.rs

```diff
@@ -734,6 +929,16 @@ where
         Ok(())
     }
 
+    fn read_16(&mut self, address: u64, data: &mut [u16]) -> Result<(), ArmError> {
+        if data.len() == 2 {
```

Comment:
Why is this 2, and why do we read a single word when the user wants 2?

Reply:
Same comment like below, should the branching be done in the inherent impl so calls to that can benefit?

Reply:
mistake, fixed